pkgs/build-support/dotnet/fetch-nupkg/overrides.nix

```diff
@@ -17,6 +17,41 @@
   #     buildInputs = old.buildInputs or [ ] ++ [ hello ];
   #   });
 
+  "Avalonia" =
+    package:
+    package.overrideAttrs (
+      old:
+      let
+        # These versions have a runtime error when built with `dotnet publish --no-build`
+        # When attempting to draw a window, Avalonia will throw "No precompiled XAML found"
+        #
+        # Introduced in https://github.com/AvaloniaUI/Avalonia/pull/13840
+        # Fixed by https://github.com/AvaloniaUI/Avalonia/pull/16835
+        affectedVersions = [
```

Comment:
That was how I originally implemented it (see https://github.com/NixOS/nixpkgs/pull/331150#discussion_r1749031290), however I came to the following conclusions:

- it would be fragile if the fix is backported
  (it's labeled as a "backport candidate")
- the number of affected versions is small enough to list
- the "beta" and "rc" versions are handled poorly by `compareVersions`
  e.g. "1.0.0" evaluates to older than "1.0.0-beta1"